From e47ac9b71290ae57d08335c88a5c9b26b5fdfe32 Mon Sep 17 00:00:00 2001
Message-Id: <e47ac9b71290ae57d08335c88a5c9b26b5fdfe32.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 18 Oct 2014 15:59:54 +0530
Subject: [PATCH 03/75] MUST_REBASE [VPG]: drm/i915/adf: Add module parameter
 to enable intel ADF.

Add a module parameter to enable intel adf. ADF support is relatively
new, and so it'd be wise to be able to switch back to i915 display,
Enable ADF once all the issue are resolved :)

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Ib3ffd85800f7b2f4d04f7a15538f88d685333cdb
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h    |    1 +
 drivers/gpu/drm/i915/i915_params.c |    5 +++++
 2 files changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 7f92936..ac0789f 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -2363,6 +2363,7 @@ struct i915_params {
 	bool prefault_disable;
 	bool reset;
 	bool disable_display;
+	bool enable_intel_adf;
 	bool disable_vtd_wa;
 	int drrs_interval;
 	int use_mmio_flip;
diff --git a/drivers/gpu/drm/i915/i915_params.c b/drivers/gpu/drm/i915/i915_params.c
index c92e804..e5ca774 100644
--- a/drivers/gpu/drm/i915/i915_params.c
+++ b/drivers/gpu/drm/i915/i915_params.c
@@ -57,6 +57,7 @@ struct i915_params i915 __read_mostly = {
 	.ring_reset_min_alive_period = 0,
 	.gpu_reset_min_alive_period = 0,
 	.memtrack_debug = 1,
+	.enable_intel_adf = 0,
 };
 module_param_named(limitbw, i915.limitbw, int, 0400);
 MODULE_PARM_DESC(limitbw,
@@ -172,6 +173,10 @@ MODULE_PARM_DESC(invert_brightness,
 module_param_named(disable_display, i915.disable_display, bool, 0600);
 MODULE_PARM_DESC(disable_display, "Disable display (default: false)");
 
+module_param_named(enable_intel_adf, i915.enable_intel_adf, bool, 0600);
+MODULE_PARM_DESC(enable_intel_adf,
+		"Override intel adf (default: 0) (0=disabled, 1=enabled)");
+
 module_param_named(disable_vtd_wa, i915.disable_vtd_wa, bool, 0600);
 MODULE_PARM_DESC(disable_vtd_wa, "Disable all VT-d workarounds (default: false)");
 
-- 
1.7.9.5

